tests/accel/upgrade/tests/upgrade.go

```diff
@@ -0,0 +1,177 @@
+			Expect(hasVersion).To(BeTrue())
+
+			By("Check that no cluster operator is progressing")
+			cosStoppedProgressing, err := clusteroperator.
+				WaitForAllClusteroperatorsStopProgressing(accelinittools.HubAPIClient, time.Minute*5)
+			Expect(err).NotTo(HaveOccurred(), "error while waiting for cluster operators to stop progressing")
+			Expect(cosStoppedProgressing).To(BeTrue(), "error: some cluster operators are still progressing")
+
+			By("Check that all cluster operators are available")
+			cosAvailable, err := clusteroperator.
+				WaitForAllClusteroperatorsAvailable(accelinittools.HubAPIClient, time.Minute*5)
+			Expect(err).NotTo(HaveOccurred(), "error while waiting for cluster operators to become available")
+			Expect(cosAvailable).To(BeTrue(), "error: some cluster operators are not available")
+
+			By("Check that all pods are running in workload namespace")
+			workloadPods, err := pod.List(accelinittools.HubAPIClient, accelparams.TestNamespaceName)
+			Expect(err).NotTo(HaveOccurred(), "error listing pods in workload namespace %s", accelparams.TestNamespaceName)
+			Expect(len(workloadPods) > 0).To(BeTrue(),
+				"error: found no running pods in workload namespace %s", accelparams.TestNamespaceName)
+
+			for _, workloadPod := range workloadPods {
+				err := workloadPod.WaitUntilReady(time.Minute * 2)
+				Expect(err).To(BeNil(), "error waiting for workload pod to become ready")
+			}
+
+			verifyWorkloadReachable(workloadRoute)
+		})
+	})
+})
+
+func startTestWorkloadAndGetRoute() *route.Builder {
+	By("Check if workload app namespace exists")
+
+	if _, err := namespace.Pull(accelinittools.HubAPIClient, accelparams.TestNamespaceName); err == nil {
+		deleteWorkloadNamespace()
+	}
+
+	By("Create workload app namespace")
+
+	_, err := namespace.NewBuilder(accelinittools.HubAPIClient, accelparams.TestNamespaceName).Create()
+	Expect(err).NotTo(HaveOccurred(), "error creating namespace for workload app")
+
+	By("Create workload app deployment")
+
+	_, err = CreateWorkload(accelinittools.HubAPIClient, accelparams.IbuWorkloadImage)
+	Expect(err).ToNot(HaveOccurred(), "error creating workload application")
+
+	By("Create workload app service")
+
+	_, err = CreateService(accelinittools.HubAPIClient, accelparams.ServicePort)
+	Expect(err).ToNot(HaveOccurred(), "error creating workload service %v", err)
+
+	By("Create workload app route")
+
+	workloadRoute, err := CreateWorkloadRoute(accelinittools.HubAPIClient)
+	Expect(err).ToNot(HaveOccurred(), "error creating workload route %v", err)
+
+	verifyWorkloadReachable(workloadRoute)
+
+	return workloadRoute
+}
+
+func deleteWorkloadNamespace() {
+	By("Delete workload")
+
+	err := DeleteNamespace(accelinittools.HubAPIClient)
+	Expect(err).NotTo(HaveOccurred(), "error deleting workload namespace %v", err)
+}
+
+func verifyWorkloadReachable(workloadRoute *route.Builder) {
+	By("Verify workload is reachable")
+
+	err := wait.PollUntilContextTimeout(
```

Comment:
Please replace this function by native ginkgo Eventually.
Check example: 
https://github.com/openshift-kni/eco-gotests/blob/37fcb7b603345b95948c9b0fd316076a9a2b24ba/tests/cnf/core/network/cni/tests/tap/tap.go#L473